### PR TITLE
First try at a configuration

### PR DESCRIPTION
**Pull request by idanstark42**
Lets start there and see what should be changed, to get this moving for Tal to start working with it.

### CODE REVIEW

File: module-standard/v1.0-SNAPSHOT.md [hunk 1]
```diff
@@ -159,7 +159,36 @@ A process must not add any option to his mongo client other then this connection
...
+      },
+      {
+        "field": "$FIELD_NAME$",
+        "values": ["$POSSIBLE_VALUE_1$", "$POSSIBLE_VALUE_2$", "$POSSIBLE_VALUE_3$"]
+      }
+    ]
+  }
+}
+```
+The configuration module will the display the fields for the user to see, grouped by their `fields-family`. A field must contain either a `type` property or a `values` property, but not both. `$FIELD_TYPE$` values can be `short text`, `long text`, `integer`, `decimal`, `date`, `datetime`, `boolean`.
```
Comment (2roy999) [inline module-standard/v1.0-SNAPSHOT.md]: lets merge short and long text, and instead add fields: `min`, `max`, `max_length`, `min_length` which would be optional.

File: module-standard/v1.0-SNAPSHOT.md [hunk 2]
```diff
@@ -159,7 +159,36 @@ A process must not add any option to his mongo client other then this connection
...
+1. `config:set_field` - when the field is changed, either by the module or by the centrelized configuration UI, the updating party must send an Mhub message `config:set_field` with the data to update the other modules. The data of this message must follow this scheme:
```
[the remark this answers is not in the document]
Comment (idanstark42) [reply module-standard/v1.0-SNAPSHOT.md]: Good point. I'll change that

File: module-standard/v1.0-SNAPSHOT.md [hunk 3]
```diff
@@ -159,7 +159,36 @@ A process must not add any option to his mongo client other then this connection
...
+```json
+{
+  "module": "$MODULE_NAME$",
+  "field-family" "$FIELD_FAMILY_NAME$",
```
[the remark this answers is not in the document]
Comment (idanstark42) [reply module-standard/v1.0-SNAPSHOT.md]: I'll fix that

File: module-standard/v1.0-SNAPSHOT.md [hunk 4]
```diff
@@ -159,7 +159,36 @@ A process must not add any option to his mongo client other then this connection
 
 ## Configuration
 
-_**TODO**_
+The system has a central UI for managing user controlled configuration. If a service wants to apply to this change it has to subscribe to the Mhub topic family `config:*` and listenr to any messages on this topic. The username for this topic family will be `configuration` and the password will be the `SECRET` environment variable. The types of messages on the `config:*` topic family will be:
+1. `config:get_fields` - a request sent by the configuration service to ask for the fields each service wishes to subscribe. This message has no data.
+1. `config:fields` - after the `config:get_fields` is sent, the module must send the fields it wishes to allow the user to configure, arranged into different `fields-families` in a `config:fields` message. The data of this message must follow this scheme:
```
[the remark this answers is not in the document]
Comment (idanstark42) [reply module-standard/v1.0-SNAPSHOT.md]: How will the configuration service access other services' `module.yml` to know what to present?

File: module-standard/v1.0-SNAPSHOT.md (hunk 1 above)
Comment (idanstark42) [reply module-standard/v1.0-SNAPSHOT.md]: I went with the same types as `<input>` tag

Comment (idanstark42) [reply module-standard/v1.0-SNAPSHOT.md]: The skirt and long text are for `text field` and `text area`

File: module-standard/v1.0-SNAPSHOT.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (2roy999) [reply module-standard/v1.0-SNAPSHOT.md]: I thought the configuration would be inside the launcher.

[the remark this answers is not in the document]
Comment (idanstark42) [reply module-standard/v1.0-SNAPSHOT.md]: Ok, I'm good with that. I'll change it

File: module-standard/v1.0-SNAPSHOT.md [hunk 5]
```diff
@@ -159,7 +159,72 @@ A process must not add any option to his mongo client other then this connection
...
+
+The message on the `config:*` topic family will be `config:update`. When the field is changed, either by the module or
+by the centrelized configuration module, the updating party must send an mhub message on this topic. The data of this
+message must follow this scheme:
+```json
+{
+  "module": "$MODULE_NAME$",
+  "fields": [
+    {
+      "name": "$FIELD_NAME$",
```
Comment (idanstark42) [inline module-standard/v1.0-SNAPSHOT.md]: There should be a group name here too, so that a group will serve as a namespace for the fields described in it

Comment (2roy999) [reply module-standard/v1.0-SNAPSHOT.md]: I thought about it, but I think that groups are mainly a UI thing so I don't want to bind between them and messages. I come to believe that a module would have 3 or 4 groups at max (one of them probably would be without a name) so i don't really see any reason to create namespaces beyond the module name.

Comment (idanstark42) [reply module-standard/v1.0-SNAPSHOT.md]: My main concern is that multiple modules will use the same field name, and when the update message is sent, they're all subscribed to the topic, and therefore will all receive it

Comment (2roy999) [reply module-standard/v1.0-SNAPSHOT.md]: The module  name is the name space. Multiple modules shouldn't share any data between them (at least in the firsts versions)

Comment (idanstark42) [reply module-standard/v1.0-SNAPSHOT.md]: Ok, I don't like that we restrict ourselves, but we can change it later if we see we need it

Comment (idanstark42) [reply module-standard/v1.0-SNAPSHOT.md]: I would press "approve"of I could, but I can't, in my opinion we can merge this

**Pull request merged**